metabase/metabase
# Use string/contains for new high cardinality string field filters

Epic https://github.com/metabase/metabase/issues/29881

For string fields with fingerprint info, we can default to `string/contains` if there are more than 20 unique values.

How to verify:
- New -> SQL query -> `SELECT * FROM PEOPLE WHERE {{filter}}`
- Change the type from `Text` to `Field Filter` -> `People` -> `Name`
- Do the same again with `People` -> `Source`. It should be just `String` because it's a low cardinality field.

<img width="935" alt="Screenshot 2023-04-07 at 14 33 46" src="https://user-images.githubusercontent.com/8542534/230602098-71c3183f-c119-49cb-88a7-8a8281022aa2.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29885)
